### PR TITLE
CI: Use official Flatpak actions

### PR DESCRIPTION
**Pull request by GeorgesStavracas**
### Description

The actions from Bilal were moved [to the Flatpak organization in GitHub](https://github.com/flatpak/flatpak-github-actions/), so let's just use that.

### Motivation and Context

Routine cleanup.

### How Has This Been Tested?

 - Let CI pass

### Types of changes

 - Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by GeorgesStavracas** [comment]
Draft because it doesn't work right now

File: .github/workflows/flatpak.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
     env:
       FLATPAK_BUILD_PATH: flatpak_app/files/share
     container:
-      image: bilelmoussaoui/flatpak-github-actions:kde-6.3
+      image: flatpak/flatpak-github-actions:kde-6.3
```
Comment (tytan652) [inline .github/workflows/flatpak.yml:53]: They still didn't switch to github registry for container so it's still bilelmoussaoui for containers.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -399,7 +399,7 @@ jobs:
           echo "OBS_GIT_TAG=$(git describe --tags --abbrev=0)" >> $GITHUB_ENV
 
       - name: Build Flatpak Manifest
-        uses: bilelmoussaoui/flatpak-github-actions/flatpak-builder@v4
+        uses: flatpak/flatpak-github-actions/flatpak-builder@master
```
Comment (tytan652) [inline .github/workflows/main.yml:402]: Switching to master is a bad idea, we should stick with v4.

File: .github/workflows/flatpak.yml [hunk 3]
```diff
@@ -68,7 +68,7 @@ jobs:
           echo "::set-output name=commitHash::$(git rev-parse --short=9 HEAD)"
 
       - name: Build Flatpak Manifest
-        uses: bilelmoussaoui/flatpak-github-actions/flatpak-builder@master
+        uses: flatpak/flatpak-github-actions/flatpak-builder@master
```
Comment (tytan652) [inline .github/workflows/flatpak.yml:71]: This one should maybe be reverted to v4.

**Review by tytan652 — COMMENTED** [review]
bilelmoussaoui's is the official Flatpak actions because it now's redirect to the flatpak repo, it's more a avoid repo redirection PR than a use official repo.

But container side no change should be done, because they still didn't switch registry.

File: .github/workflows/flatpak.yml (hunk 1 above)
Comment (pheiduck) [reply .github/workflows/flatpak.yml:53]: They didn’t update there README https://github.com/flatpak/flatpak-github-actions so change is correct

File: .github/workflows/main.yml (hunk 2 above)
Comment (pheiduck) [reply .github/workflows/main.yml:402]: And yes, for stability version tags are better

File: .github/workflows/flatpak.yml (hunk 3 above)
Comment (pheiduck) [reply .github/workflows/flatpak.yml:71]: Same here

File: .github/workflows/flatpak.yml (hunk 1 above)
Comment (derrod) [reply .github/workflows/flatpak.yml:53]: No the change is incorrect. The docker image is still not published on the flatpak docker hub or github packages (https://hub.docker.com/u/flatpak), only on https://hub.docker.com/u/bilelmoussaoui

Comment (tytan652) [reply .github/workflows/flatpak.yml:53]: > They didn’t update there README https://github.com/flatpak/flatpak-github-actions so change is correct

The pull request to switch to Github registry as `flatpak` is not merge, for now it's in docker registry as `bilelmoussaoui`.

**Comment by GeorgesStavracas** [comment]
I'm aware of all these issues folks, and I'm working on fixing them in https://github.com/flatpak/flatpak-github-actions - this is why this PR is a draft in the first place

**Pull request closed without merging**